tools/executors/directus/impl.js

```diff
@@ -174,7 +174,10 @@ function runAction(options, context) {
           directus.on('error', function (error) {
             reject(error)
           })
-          directus.on('exit', function () {
+          directus.on('exit', function (code) {
+            if (code !== 0) {
+              reject(code)
```

Comment:
I _think_ we need a `return` statement here too.

```suggestion
              reject(code)
              return
```

Reply:
yep turns out. the promise stays rejected but it'd still execute any other code afterward if not returned, never really even considered that!

Reply:
Now that I think about it, another option is to wrap the resolve in an else block.